docling/backend/msword_backend.py

```diff
@@ -1223,7 +1259,25 @@ def _handle_tables(
                 else:
                     text = text.replace("<eq>", "$").replace("</eq>", "$")
 
-                table_cell = TableCell(
+                provs_in_cell: List[RefItem] = []
+                _, provs_in_cell = self._walk_linear(cell._element, docx_obj, doc)
+                ref_for_rich_cell = provs_in_cell[0]
+                group_name = f"rich_cell_group_{len(doc.tables)}_{col_idx}_{row.grid_cols_before + row_idx}"
+                group_element = doc.add_group(
+                    label=GroupLabel.UNSPECIFIED,
+                    name=group_name,
+                    parent=docling_table,
+                )
+                for prov in provs_in_cell:
+                    group_element.children.append(prov)
+                    pr_item = prov.resolve(doc)
+                    item_parent = pr_item.parent.resolve(doc)
+                    if pr_item.get_ref() in item_parent.children:
+                        item_parent.children.remove(pr_item.get_ref())
+                    pr_item.parent = group_element.get_ref()
+                ref_for_rich_cell = group_element.get_ref()
+
+                table_cell = RichTableCell(
```

Comment:
Do we think all table-cells will be always of type `RichTableCell`, or could we envision a way to detect easily if something is just a `TableCell`. 

This is just an open question.